exporter/datadogexporter/internal/clientutil/api.go

```diff
@@ -46,6 +46,9 @@ func ValidateAPIKey(ctx context.Context, apiKey string, logger *zap.Logger, apiC
 	}
 	if err != nil {
 		logger.Warn("Error while validating API key", zap.Error(err))
+		if httpresp.StatusCode == 403 {
```

Comment:
Needs to fix test: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/9838873873/job/27159740034?pr=33944#step:8:302